crates/criticalup-core/src/download_server_client.rs

```diff
@@ -71,9 +72,23 @@ impl DownloadServerClient {
         product: &str,
         release: &str,
     ) -> Result<ReleaseManifest, Error> {
-        let p = format!("/v1/releases/{product}/{release}");
+        // TODO: tests
```

Comment:
Is this blocking this PR or for the future?

Reply:
For the future. I did add one measly test but more can be done once I consolidate the test utils.